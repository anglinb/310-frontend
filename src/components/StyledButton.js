import React from 'react'
import {
  View,
  Text,
  Button,
  StyleSheet,
  TouchableHighlight,
} from 'react-native'

import { LinearGradient } from 'expo';

import config from '../config'

export default class StyledButton extends React.Component {

  constructor(props) {
    super(props)
  }

  render() {
    return (
      <TouchableHighlight onPress={this.props.onPress}>
        <View>
          <LinearGradient
            colors={config.greenGradient}
            style={buttonStyles.gradient}>
            <Text
              style={buttonStyles.text}>
              { this.props.title }
            </Text>
          </LinearGradient>
        </View>
      </TouchableHighlight>
    )
  }
}

const buttonStyles = StyleSheet.create({
  text: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#fff',
    backgroundColor: 'transparent',
    fontFamily: config.font 
  },
  gradient: {
    padding: 15,
    alignItems: 'center', 
    borderRadius: 5
  }
})