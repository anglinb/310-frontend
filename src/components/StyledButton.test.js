import React from 'react';
import StyledButton from './StyledButton';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const rendered = renderer.create(
    <StyledButton
      title={`Winning`}
      onPress={() => {
        console.log('eyu')
      }}
     />
  ).toJSON()
  expect(rendered).toBeTruthy()
  expect(rendered).toMatchSnapshot()
});
