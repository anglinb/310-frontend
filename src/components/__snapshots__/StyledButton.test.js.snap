// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <ExponentLinearGradient
      colors={
        Array [
          4278628033,
          4278632359,
        ]
      }
      end={undefined}
      locations={null}
      start={undefined}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "padding": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "Avenir Next",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Winning
      </Text>
    </ExponentLinearGradient>
  </View>
</View>
`;
